kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -553,6 +553,18 @@
       }
     },
     methods: {
+      /**
+       * Uses the imported `hasCheckbox` method in addition to some locally relevant conditions
+       * to identify if the content has a checkbox.
+       * Note that `hasCheckbox` handles the topic-level logic and we're only modifying how we
+       * handle the case of exercises where we want to show the checkbox if ith as no questions
+       * available
+       */
+      actuallyHasCheckbox(content) {
+        return content.kind === ContentNodeKinds.EXERCISE
+          ? this.hasCheckbox(content) && this.unusedQuestionsCount(content) > 0
+          : this.hasCheckbox(content);
```

Comment:
Note that we're assuming here that the API is the thing that matches our given string where we say a folder can only have <=12 resources in it and no sub-folders.

I'm thinking we maybe should be enforcing that on the client-side in order to be sure we will always match our string

Reply:
`hasCheckbox` is saying a topic cannot have a checkbox if there is a `more` property on the response -- that is where the assumption is made against the API in `useQuizResources`

https://github.com/nucleogenesis/kolibri/blob/463798f2fc4d2fab25ccd5b6d7fc1056795be607/kolibri/plugins/coach/assets/src/composables/useQuizResources.js#L139-L140